svc/ChangelogService.js

```diff
@@ -0,0 +1,205 @@
+ * into JSON and make available via the special `@xh/app-changelog.json` import above.
+ *
+ * Alternatively, changelog entries can be supplied via an optional `xhChangelog` JSON app-config
+ * matching the format of the {@see Changelog} typedef below, with the config taking precedence.
+ *
+ * If a changelog is available, the top-level app menu will include an item to view the log in a
+ * dialog. If a string pref is defined with key `xhLastReadChangelog`, this service will track the
+ * most recent version viewed by the user, and the app will render a "What's new?" button in the
+ * top appBar to alert the user. These app-level UI elements are currently desktop-only, can all be
+ * independently disabled, and only appear when this service is enabled.
+ *
+ * Several additional options can be controlled via soft-config - see below.
+ *
+ * @see XH.showChangelog - public API for displaying the changelog, if enabled and populated.
+ * @see whatsNewButton - utility button that conditionally renders when an unread entry exists for
+ *      the currently deployed app version. Installed by default in desktop appBar.
+ */
+export class ChangelogService extends HoistService {
+
+    // Optional JSON AppConfig key to soft-configure this service - see this.config for shape.
+    SVC_CONFIG_KEY = 'xhChangelogConfig';
+    // Optional JSON AppConfig key to maintain or override the log entirely via config.
+    LOG_CONFIG_KEY = 'xhChangelog';
+    // Optional string Preference key to track last read log entry + enable unread notifications.
+    LAST_READ_PREF_KEY = 'xhLastReadChangelog';
+
+    /**
+     * @member {Changelog} - the complete changelog object, as produced by the app build or defined
+     *      in soft-config, or an empty placeholder if neither source is enabled/available.
+     */
+    changelog;
+
+    /**
+     * @member {ChangelogVersion[]} - parsed and cleaned versions, if any, with any versions or
+     *      nested categories marked for exclusion via config already omitted.
+     */
+    versions;
+
+    /**
+     * @member {boolean} - true if an entry exists for the current app version which the user has
+     *      yet to view (and the required user preference has been created).
+     */
+    @observable currentVersionIsUnread;
+
+    /** @return {boolean} */
+    get enabled() {return XH.isDesktop && this.config.enabled && !isEmpty(this.versions)}
+
+    /** @return {?string} */
+    get latestAvailableVersion() {
+        if (!this.enabled) return null;
+        return this.versions[0].version;
+    }
+
+    /** @return {?Object} */
+    get currentVersionEntry() {
+        if (!this.enabled) return null;
+        return this.versions.find(it => it.version === XH.appVersion) ?? null;
+    }
+
+    constructor() {
+        super();
+        makeObservable(this);
+    }
+
+    async initAsync() {
+        const jsonFromConfig = XH.getConf(this.LOG_CONFIG_KEY, {});
```

Comment:
I am not sure I get why we are supporting this alternative way of authoring.  Its a real complexifier, and also seems like it would tie us to supporting a particular json format of the markdown in future implementations.

Reply:
Are we loading this data from when it is disabled e.g. mobile?  Seems confusing...